src/smallestPrimeFactor.js

```diff
@@ -12,14 +12,61 @@
  * @returns {number} The smallest prime factor of num
  */
 function smallestPrimeFactor(num) {
-  // This is your job. :)
+  /* const primeFactors = [];
+  let workingNum = num;
+  for (let n = 2; n <= num/2; n++) {
+    while (workingNum % n === 0) {
+      primeFactors.push(n);
+      workingNum /= n;
+    }
+
+  } */
+
+  for (let n = 2; n <= num; n++) {
```

Comment:
What's the largest the smallest prime factor can be? It's smaller than `num`.

Consider a number like `18`. Will we ever check if `9` is a factor? No, because if `9` is a factor then so is `18 / 9 === 2`. Since `2 < 9`, we would have already checked `2` and returned.

Likewise, if, say, `17` had a proper factor `k` then both `k` and `17 / k` are proper factors. If `k` is not a factor and `17 / k > k` then we don't even have to bother checking for `17 / k` because we would've already seen that `k` was a factor.

Since `17` is prime, there's no such `k`. For example, `17 / 5` is `3.4`. If `5` were a factor then we would've already checked for its complementary factor. Since we haven't returned by the time we're checking `5`, we know there aren't any possible prime factors to find.

What's the largest that the smallest prime factor can be? We only need to iterate up to that point.